File: SPA_QA_oblig3/Models/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace SPA_QA_oblig3.Models
{
    public class Question
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public String Asked { get; set; }

        [Required]
        public Category Category { get; set; }

        [Required]
        public String AskedBy { get; set; }

        [Required]
        public String Email { get; set; }

        public String Answer { get; set; }

        public bool Answered { get; set; }
        
        public bool IsFAQ { get; set; }



    }
}

File: SPA_QA_oblig3/App/Question.ts
import { Category } from "./Category";

export class Question {
    id: number;
    asked: string;
    answered: boolean;
    answer: string;
    category: Category;
    askedBy: string;
    isFAQ: boolean;
    email: String;

    constructor(id: number, asked: string, answered: boolean, answer: string, category: Category, askedBy: string, isFAQ: boolean, email: String) {
        this.id = id;
        this.asked = asked;
        this.answered = answered;
        this.answer = answer;
        this.category = category;
        this.askedBy = askedBy;
        this.isFAQ = isFAQ;
        this.email = email;

        if (this.answer == "") {
            this.answer = "NOT ANSWERED";
        }
    }
}